src/System.Private.Xml/src/System/Xml/XmlDownloadManager.cs

```diff
@@ -36,22 +34,26 @@ internal partial class XmlDownloadManager
         private Stream GetNonFileStream(Uri uri, ICredentials credentials, IWebProxy proxy,
             RequestCachePolicy cachePolicy)
         {
-            HttpClient client = new HttpClient();
-            HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Get, uri);
-
-            lock (this)
+            WebRequest req = WebRequest.Create(uri);
+            if (credentials != null)
             {
-                if (_connections == null)
-                {
-                    _connections = new Hashtable();
-                }
+                req.Credentials = credentials;
+            }
+            if (proxy != null)
+            {
+                req.Proxy = proxy;
+            }
+            if (cachePolicy != null)
+            {
+                req.CachePolicy = cachePolicy;
             }
 
-            HttpResponseMessage resp = client.SendAsync(req).GetAwaiter().GetResult();
-
-            Stream respStream = new MemoryStream();
-            resp.Content.CopyToAsync(respStream);
-            return respStream;
+            using (WebResponse resp = req.GetResponse())
+            {
+                MemoryStream respStream = new MemoryStream();
+                resp.GetResponseStream().CopyToAsync(respStream);
```

Comment:
If this is working, it's only by accident (same goes for the old code).  This copy needs to have finished by the time you return the stream, otherwise the stream could end up getting used concurrently by both the copy and the caller.  My guess is that at least in the tested situations the type of GetResponseStream is a MemoryStream, such that its CopyToAsync ends up being entirely synchronous, but if it's something else that would result in asynchronous copies, this would be very buggy.  If this operation needs to be synchronous, you should use CopyTo rather than CopyToAsync here.